bindings/python/src/decoders.rs

```diff
@@ -661,6 +643,13 @@ impl PyDecodeStream {
         }
     }
 
+    #[pyo3(signature = (sequence_ids), text_signature = "(self, sequence_ids)")]
+    fn with_sequence(&mut self, sequence_ids: Vec<u32>) {
+        self.ids = sequence_ids;
+        self.prefix_index = self.ids.len();
+        self.prefix = "".to_string();
```

Comment:
You're trashing the prefix here, I'm not sure how that'll play with the prefix spaces. I think this will effectively reset the start which might be odd.

Have you thought about simply creating a new DecodeStream from a vec of ids (making it extra clear about the intent?)

Reply:
Did not and it does make sense, will try that